openedx_tagging/core/tagging/models/base.py

```diff
@@ -310,6 +314,37 @@ def get_tags(self, tag_set: models.QuerySet | None = None) -> list[Tag]:
                 break
         return tags
 
+    def get_filtered_tags(
+        self,
+        tag_set: models.QuerySet[Tag] | None = None,
+        parent_tag_id: int | None = None,
+        search_term: str | None = None,
+        search_in_all: bool = False,
+    ) -> models.QuerySet[Tag]:
+        """
+        Returns a filtered QuerySet of tags.
+        By default returns the root tags of the given taxonomy
+
+        Use `parent_tag_id` to return the children of a tag.
+
+        Use `search_term` to filter the results by values that contains `search_term`.
+
+        Set `search_in_all` to True to make the search in all tags on the given taxonomy.
+        """
+        if tag_set is None:
+            tag_set = self.tag_set.all()
+
+        if self.allow_free_text:
+            return tag_set.none()
```

Comment:
Do you mean having two functions? `get_tags`  and `get_filtered_tags`?

Reply:
No I mean for example that it would be nice if `search_tags` worked for any taxonomy, but right now it only works for taxonomies that aren't free text. We can change that in the future though; no change for now. After all we have excluded free text from the MVP anyways.